### ISSUE TITLE
Update for nullability changes in 15.0.0-p.12

### ISSUE THREAD

**Issue by cmeeren** [cmeeren/FSharp.HotChocolate#20]
@michaelstaib, I'm having problems due to [this change](https://github.com/ChilliCream/graphql-platform/pull/7619/files#diff-ff3832c0b3aa1c6f30522727cd20adc75bfb887562eb1221489df6bbf8fc7730) in 15.0.0-p.12, which removed some `Option<_>` unwrapping I was not aware was present in the core of HotChocolate.

I am trying to modify FSharp.HotChocolate to account for this (not committed/pushed yet). I am now removing `Option` wrappers myself, so now the field/argument types are correct (and correctly queryable), but the schema is still full of unused/unreachable types like this:

```graphql
type FSharpOptionOfDouble {
  get_Value: Float!
  value: Float!
}
```

Again, **these are unused/unreachable**. For example, this particular type appears if I have a field that returns `Option<double>` (which, in the schema, is correctly typed as `Float`, not `FSharpOptionOfDouble`).

Why is that happening, and what can I do about it? (Apart from `.TrimTypes()`, which I obviously don't want to force on our users, and which doesn't fix the root issue.)

**Comment by michaelstaib** [comment]
Yes, we removed FSharp related code from the core as with your package we do not see need for ours. However we will have new hooks that you can tap into the core. However, at the moment we are just doing some realignment for the main project, so we kick things out but have not yet put any new things in.

**Comment by cmeeren** [comment]
Thanks for the quick reply! So there is nothing for me to do except wait for p.13 (or whichever release this will be present in)?

Could I bother you to post a comment here when and how I can use the new API?

**Comment by michaelstaib** [comment]
I still have to think about how we redesign the type inspection. We need to prepare the type inspection to allow for providers as we also want to use source generators for C# here, but for F# you might want to stay on the reflection API or tap into F# source generators once they arrive,

**Comment by michaelstaib** [comment]
Yes, nothing todo yet, I will ping you once we untangled the type inspection.

**Comment by cmeeren** [comment]
Hi, now that v15 is out, could you help me with FSharp.HotChocolate? Currently it's completely broken, as described in the OP (hopefully not a complicated fix, but I don't know what).

**Comment by cmeeren** [comment]
Hi @michaelstaib, just a reminder about this. 🙂 I'm dependent on your help and that "knowledge transfer" you have talked about previously, in order to update this library for v15, so that F# users are able to use HC 15.

**Comment by cmeeren** [comment]
This is now supported.

References:
https://github.com/ChilliCream/graphql-platform/issues/8088
https://github.com/ChilliCream/graphql-platform/pull/8722
https://github.com/ChilliCream/graphql-platform/pull/8732

**Comment by mta-incom** [comment]
@cmeeren Any updates on a release of this issue?
As far as i can see the changes required in HC were added in [HC 15.1.11](https://github.com/ChilliCream/graphql-platform/releases/tag/15.1.11)

But the fix for it in FSharp.HotChocolate (here: https://github.com/cmeeren/FSharp.HotChocolate/commit/763d1842f46507ddfc95251aec229f9536e5b887) has not yet been released and the change targets a pre-release candidate (16.0.0-p.8.4) of HC rather than the latest stable.

Any updates would be much appreciated.

**Comment by cmeeren** [comment]
I'm waiting for the non-prerelease HC 16.0. When that is out, I can update and release FSharp.HotChocolate (which will then require HC 16).

**Comment by IkertxoDt** [comment]
Hello

We have a fresh new version https://www.nuget.org/packages/HotChocolate/16.0.0

Maybe do you have and estimated time for the relase of this fix? 

Thanks in advanced!

**Comment by cmeeren** [comment]
Finally! 🙂 I have a lot on my plate at the moment, but I'll get to it as soon as I'm able.

**Comment by IkertxoDt** [comment]
I hope this helps! I was able to download the latest version from the **main** branch and here is what I found:

* I changed the `HotChocolate.Execution` reference to `HotChocolate` because `HotChocolate.Execution` v16 does not exist. I’m not sure if this is the correct solution, but it worked for me.
* I updated all *NuGets* references to their latest versions.
* Everything compiles correctly except for the `MyUnion4Descriptor` type in the `UnionsAsUnions.fs` file, as there were *no members to override*. I have commented out this part and the dependent code.
* Tests pass, except that the GraphQL definition file changes significantly (e.g., the order of the members). It will be necessary to check the snapshot tests very carefully.

I updated my project with the resulting DLL and updated HotChocolate to v16.

For my tests, I use [ZeroQL](https://github.com/byme8/ZeroQL) to generate a C# client from the GraphQL file. Although the file structure changed a lot, the resulting client is the same, so I didn't have to change a single line of code.

I ran the tests... and they all passed!

I think you have done **great work** and everything is working as expected. **I predict a great new version!** :)

**Comment by cmeeren** [comment]
Thanks, but I've already done it locally. Getting in some additional fixes before I push and release.

**Comment by cmeeren** [comment]
[v1.0.0](https://github.com/cmeeren/FSharp.HotChocolate/releases/tag/v%2F1.0.0) released now. 🚀

**Comment by IkertxoDt** [comment]
¡¡¡Congratulations!!!  And thanks a lot. Just downloaded and testing it :)
